generator/prompt_generator.py

```diff
@@ -0,0 +1,92 @@
+import re
+import os
+
+from loguru import logger
+from huggingface_hub import login
+
+import generator.utils.io_utils as io_utils
+from generator.backend_generators.groq_generator import GroqGenerator
+from generator.backend_generators.vllm_generator import VLLMGenerator
+
+
+class PromptGenerator:
```

Comment:
I would either rename this class or rename the backend_generators to generator_backend. i.e. BaseGenerator Backend, GroqBackend, etc. Otherwise I would expect PromptGenerator and the other classes to have the same base type or implement the same interface oder generally do the same job.

Reply:
That makes sense, thank you for the advice !